### DISCUSSION TITLE
Prometheus Alertmanager: custom headers (bearer token)

### DISCUSSION THREAD

**Discussion by nabokihms** [grafana/grafana#42806] [Alerting]
Hello again, Grafana team.

One more thing about alertmanagers. 
![image](https://user-images.githubusercontent.com/32434187/144978960-44d39f84-2b8a-429f-a6de-a2a991e17f22.png)

There is no option to configure custom headers, which limits the possibilities for authentication (we cannot use bearer tokens or other helpful headers). 
1. It is possible to add headers to the SendWebhookSync structure as I see it. Thus the only change we should make is propagation headers from the GUI to the backend.
https://github.com/grafana/grafana/blob/2af5ad4764c8f2a0c8ab223c5528ba44efde01c3/pkg/services/alerting/notifiers/alertmanager.go#L178-L184

2. The other possibility is to make only the bearer token option along with the basic auth. Then we will be able to store tokens in the secret store (like the primary auth password).

Which variant do you think is the most preferable? Is it worth adding such an option to Grafana?

**Comment by gotjosh** [comment]
Thank you for the feature request - appreciate the time taken to write in so much detail ❤️ 

I have opened https://github.com/grafana/grafana/pull/52515, which suggest #2 - Does this solve your problem?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
